demo/server/src/api/download.js

```diff
@@ -0,0 +1,82 @@
+const https = require('https');
+const aws4 = require('aws4');
+
+export const handler = (event, context, callback) => {
+  const region = process.env.REGION;
+  const apiGatewayId = process.env.SIGNED_UPLOADS_API_ID;
+  const stage = process.env.STAGE;
+  const endpoint = '/storage/download';
+  const accessKeyId = process.env.PRODUCER_KEY_ID;
+  const secretAccessKey = process.env.PRODUCER_SECRET_ACCESS_KEY;
+
+  const computedKey =
+    event.queryStringParameters && event.queryStringParameters.key
+      ? event.queryStringParameters.key
+      : undefined;
+
+  if (!computedKey) {
+    const response = {
+      statusCode: 400,
+      body: JSON.stringify({
+        message: `Missing key parameter`,
+      }),
+    };
+
+    return callback(null, response);
+  }
+
+  const params = {
+    host: `${apiGatewayId}.execute-api.${region}.amazonaws.com`,
+    method: 'GET',
+    path: `/${stage}${endpoint}`,
+    headers: {
+      'x-amz-meta-computed-key': computedKey,
+    },
+  };
+
+  // can specify any custom option or header as per usual
+  const req = https.request(
+    aws4.sign(params, {
+      accessKeyId,
+      secretAccessKey,
+    }),
+    res => {
+      res.setEncoding('utf8');
+      let body = '';
+
+      res.on('data', data => {
+        body += data;
+      });
+
+      res.on('end', () => {
+        const { statusCode } = res;
+
+        if (statusCode === 200) {
+          return callback(null, {
+            statusCode: 200,
+            headers: {
```

Comment:
Maybe we can prepare the response and add status code and body correspondingly to avoid the repetition..